### ISSUE TITLE
Picker to Page

### ISSUE THREAD

**Issue by pixeljunker** [AmomentOfMusic/Zotero_scrivener_picker_windows#3]
I can use the picker, I think, as in the formatting works, I can select my citations, but then I have no idea how to move the citation from the picker (box) into the scrivener file. Copying and pasting isn't a thing it seems, hitting enter results in a green progress bar but no citation in-line. I appreciate you putting the work in, great script idea.

**Comment by AmomentOfMusic** [comment]
For me, I just call up the picker, select the citation and then hit enter - the citation is then inserted into Scrivener automatically after the progress bar. I'm not sure why it isn't working in your case, but here are some thoughts:

1. Are you using the latest version of this script? (EDIT: see comment below. TLDR; I changed the script so you no longer need Autoit installed)

2. Have you installed the ODF scan plugin? Otherwise, the scannable-cite citation style will not exist, and that might also be causing an issue. Make sure that it is properly installed - I know I had issues where I had to install the plugin manually. This was supposed to be fixed, but it might not be. You can check if it's installed by clicking in Zotero edit>export>default format>scannable cite. If Scannable Cite does not appear as an option, then that is probably the issue. [See this forum post for how to manually install](https://forums.zotero.org/discussion/57428/scannable-cite-not-showing-up-in-zotero-standalone)

3. While Scrivener (with your cursor in the editor) and Zotero are open in the background, right-click on the file "Scrivener picker" and click "edit". This will open the script in the powershell editor.  Click on the green play button. This will run the script. Once the picker pops up, select a citation, and press enter. For me, this automatically switches to Scrivener and inserts the citation. What happens for you? After you've run the script once, return to the powershell editor and look at the black portion at the bottom of the screen. Copy paste whatever text is there and post it here - that should give me an idea of what went wrong.

Finally, as a work around you can also do the following: edit the portion of the script which says 

`$ref = invoke-expression "curl 'http://localhost:23119/better-bibtex/cayw?format=scannable-cite' -usebasicparsing"`

so that it  says

`$ref = invoke-expression "curl 'http://localhost:23119/better-bibtex/cayw?format=scannable-cite&clipboard=yes' -usebasicparsing"`

Make sure to save before running the script again. The addition of the "&clipboard=yes" will copy the citation to the clipboard. You should then be able to paste the text into Scrivener. Although hopefully we can fix your problem so that you don't have to use this workaround!

**Comment by AmomentOfMusic** [comment]
I have just played around on another computer to check everything is working, and I've realized that uninstalling autoit, unlike my instruction, does indeed get ride of the cdmlet features the script needs (it didn't on my computer, probably because I had installed them manually the first time around...)

I have tweaked the script a bit so that the files are now included in the folder and autoit no longer has to be installed in addition. I'm not sure whether that is your issue or not, but if it is, perhaps try re-downloading the folder now and see if it works.

**Comment by poimaino** [comment]
Hi there, thanks so much for developing this picker!
I'm having similar issues to this user; the script brings up the picker, but when I hit enter after specifying the source I want to cite, nothing appears in Scrivener. Here's the error I'm getting from Powershell:

> Show-AU3WinActivate : The term 'Show-AU3WinActivate' is not recognized as the 
> name of a cmdlet, function, script file, or operable program. Check the 
> spelling of the name, or if a path was included, verify that the path is 
> correct and try again.
> At C:\Zoterobits\Zotpick\zotpick\Scrivener-picker.ps1:49 char:1
> + Show-AU3WinActivate($Activewindow)
> + ~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : ObjectNotFound: (Show-AU3WinActivate:String) [], 
>     CommandNotFoundException
>     + FullyQualifiedErrorId : CommandNotFoundException
>  
> Wait-AU3WinActive : The term 'Wait-AU3WinActive' is not recognized as the name 
> of a cmdlet, function, script file, or operable program. Check the spelling of 
> the name, or if a path was included, verify that the path is correct and try 
> again.
> At C:\Zoterobits\Zotpick\zotpick\Scrivener-picker.ps1:50 char:1
> + Wait-AU3WinActive($Activewindow)
> + ~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : ObjectNotFound: (Wait-AU3WinActive:String) [], C 
>    ommandNotFoundException
>     + FullyQualifiedErrorId : CommandNotFoundException
>  
> Send-AU3Key : The term 'Send-AU3Key' is not recognized as the name of a 
> cmdlet, function, script file, or operable program. Check the spelling of the 
> name, or if a path was included, verify that the path is correct and try again.
> At C:\Zoterobits\Zotpick\zotpick\Scrivener-picker.ps1:53 char:1
> + Send-AU3Key -key $ref -mode 2
> + ~~~~~~~~~~~
>     + CategoryInfo          : ObjectNotFound: (Send-AU3Key:String) [], Command 
>    NotFoundException
>     + FullyQualifiedErrorId : CommandNotFoundException
> 
```

**Comment by AmomentOfMusic** [comment]
Hi there,

From the error, it seems like the AutoIt Cmdlet is not loading probably, which is what I rely on to send keys to the window and pull windows to front (my first versions of the script did without, but it meant that sometimes the picker wouldn't come to the front)? 

I just uploaded a change I did a while back (that I realized I never committed here) so hopefully that will address the issue. If it doesn't then I would recommend putting the 5 autoitx files into a folder entitled "AutoItX" and moving it to the following directory: C:\Program Files (x86)\WindowsPowerShell\Modules\

Basically that will install autoit globally so that there shouldn't be an issue with calling the cmdlets. Let me know if that fixes your problem!

**Comment by poimaino** [comment]
Thanks so much! It looks like that (and a few other tweaks) have resolved those issues running the script. It's working!

**Comment by AmomentOfMusic** [comment]
Awesome :) Can you let me know what the tweaks you did were? If there is any info I can include in the readme to help others to troubleshoot their issues, then I'm happy to do so. 

Happy citing!

**Comment by poimaino** [comment]
Sure. Some of these may reflect idiosyncrasies in my setup - I'm not expert enough to know!

1. I had the issue you described with the shortcut not working, so had to create a new shortcut. Once I forgot to make this run the script minimized, so until I corrected that it wasn't working.
2. The AutoItX files didn't load when stored locally with the picker script, **nor** in the x86 program files folder - but they did when stored in C:\Program Files\WindowsPowershell\modules\  (i.e. NOT x86).
3. Depending on where I saved the files, I had some permissions issues running the script, which were solved either by (1) copying the code out into a new, local script file, or (2) executing an unblock command on the script. I did both, so not sure which one did it.
